ovs-service/src/main/java/org/dcsa/ovs/service/VesselScheduleService.java

```diff
@@ -1,21 +1,172 @@
+    String carrierServiceCode;
+    String universalServiceReference;
+    String vesselIMONumber;
+    String vesselName;
+    String voyageNumber;
+    String universalVoyageReference;
+    String unLocationCode;
+    String facilitySMDGCode;
+    String startDate;
+    String endDate;
+    Integer limit;
+    String cursor;
+    String apiVersion;
+  }
+
+  public List<VesselScheduleTO> findAll(ServiceSchedulesFilters requestFilters) {
+
+    return serviceRepository.findAll().stream()
+        .map(
+            service ->
+                VesselScheduleTO.builder()
+                    .carrierServiceName(service.getCarrierServiceName())
+                    .carrierServiceCode(service.getCarrierServiceCode())
+                    .universalServiceReference(service.getUniversalServiceReference())
+                    .vessels(extractVesselsToVesselTO(service.getVessels()))
+                    .build())
+        .toList();
+  }
+
+  private List<VesselTO> extractVesselsToVesselTO(List<Vessel> vessels) {
+    return vessels.stream()
+        .map(
+            vessel ->
+                VesselTO.builder()
+                    .vesselName(vessel.getName())
+                    .vesselIMONumber(vessel.getImoNumber())
+                    .vesselOperatorCarrierSMDGCode(vessel.getOperatorCarrier().getSmdgCode())
+                    .vesselCallSign(vessel.getCallSign())
+                    .vesselLength(vessel.getLength())
+                    .vesselWidth(vessel.getWidth())
+                    .dimensionUnit(
+                        DimensionUnit.valueFromString(vessel.getDimensionUnit()).orElse(null))
+                    .isDummyVessel(vessel.getIsDummy())
+                    .portCalls(extractTransportCallToPortCallTo(vessel.getPortCalls()))
+                    .build())
+        .toList();
+  }
+
+  private List<PortCallTO> extractTransportCallToPortCallTo(List<TransportCall> transportCalls) {
+    return transportCalls.stream()
+        .map(
+            transportCall ->
+                PortCallTO.builder()
+                    .transportCallReference(transportCall.getReference())
+                    .importVoyageNumber(
+                        Objects.requireNonNullElse(transportCall.getImportVoyage(), new Voyage())
+                            .getCarrierVoyageNumber())
+                    .exportVoyageNumber(
+                        Objects.requireNonNullElse(transportCall.getExportVoyage(), new Voyage())
+                            .getCarrierVoyageNumber())
+                    .importUniversalVoyageReference(
+                        Objects.requireNonNullElse(transportCall.getImportVoyage(), new Voyage())
+                            .getUniversalVoyageReference())
+                    .exportUniversalVoyageReference(
+                        Objects.requireNonNullElse(transportCall.getExportVoyage(), new Voyage())
+                            .getUniversalVoyageReference())
```

Comment:
I think these are ripe for `Objects.requireNonNullElseGet` using `Voyage::new` as second parameter.

Reply:
Alternatively a `private static final EMPTY_VOYAGE = new Voyage();` in the top instead of multiple `new Voyage()` could also do.